### ISSUE TITLE
Improve error messages for value types mismatches

### ISSUE THREAD

**Issue by patrick91** [apollographql/federation#296]
This PR adds a fix for how we check value types: https://github.com/apollographql/federation/pull/280 but the error message could be improved, for example:

```javascript
const [definitions] = createDocumentsForServices([
{
    typeDefs: gql`
    type Person {
        age(relative: Boolean!): Int
    }
    `,
    name: 'serviceA',
},
{
    typeDefs: gql`
    type Person {
        age(relative: Boolean): Int
    }
    `,
    name: 'serviceB',
},
]);

const errors = validateSDL(definitions, schema, [
    UniqueTypeNamesWithFields,
]);
```

Returns this error:

```
[serviceA] Person -> A field's input type (`relative`) was defined differently in different services. `serviceA` and `serviceB` define `relative` as a Boolean! and Boolean respectively. In order to define `Person` in multiple places, the input values and their types must be identical.
```

It would be nice to include the field name in the error, like this:

```
[serviceA] Person.age -> A field's argument (`relative`) was defined differently in different services. `serviceA` and `serviceB` define the argument `relative` on `Person.age` as a Boolean! and Boolean respectively. In order to define `Person` in multiple places, the input values and their types must be identical.
```

Link to discussion and code: https://github.com/apollographql/federation/pull/280/files#r535572567

**Comment by sarink** [comment]
If you need to change a type, how are you supposed to do it? 🤔

**Comment by patrick91** [comment]
@sarink you can extend types with federation https://www.apollographql.com/docs/federation/entities/#extending

but if you have a type that's duplicated in multiple schemas then you'd need to change this type everywhere :)

**Comment by sarink** [comment]
@patrick91 yes, you would need to change it everywhere, but how can you publish the changes?

**Comment by patrick91** [comment]
@sarink can you explain your use case a bit more? not sure what's the issue (I also don't use the apollo registry, so maybe I'm missing some pieces)

**Comment by sarink** [comment]
If two services define the same type (it's duplicated), how can you possibly change them? If you do, you'll run into the error in this PR, that the types don't match.

**Comment by glasser** [comment]
I think there's two different issues here:

(a) The current composition algorithm does a lot of helpful checking to make sure that your subgraph schemas are appropriately compatible with each other. But while the errors these checking rules generate are helpful at ensuring you don't accidentally end up with incompatible subgraphs, they aren't designed in a way that allows you to evolve your schema via a series of incremental compatible changes. We need to improve a lot of these rules so that making changes to your schema can be done safely.  This is something we've been talking about more lately at Apollo and I hope to see some progress on relatively soon, but we can also try to make some incremental progress on specific cases if we have an uncontroversial design.

(b) This particular error message doesn't say what field it's talking about. That's definitely confusing! (Also, I think instead of "input" it should say "argument" in the message.) I think a fix to improve the wording would be easy to merge if provided as a PR.
